ICAFIX/ReApplyFixMultiRunPipeline.sh

```diff
@@ -168,11 +173,63 @@ determine_old_or_new_fsl()
 have_hand_reclassification()
 {
 	local StudyFolder="${1}"
-	local Subject="${2}"
+	local Session="${2}"
 	local fMRIName="${3}"
 	local HighPass="${4}"
 
-	[ -e "${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_hp${HighPass}.ica/HandNoise.txt" ]
+	[ -e "${StudyFolder}/${Session}/MNINonLinear/Results/${fMRIName}/${fMRIName}_hp${HighPass}.ica/HandNoise.txt" ]
+}
+
+# ------------------------------------------------------------------------------
+# Copy necessary files from cross-sectional to longitudinal target
+# ------------------------------------------------------------------------------
+function copy_to_longitudinal()
+{
+	local StudyFolder="$1" SessionCross="$2" SessionLong="$3" fMRINames="$4" ConcatName="$5" HighPass="$6"
+	local fmri S T
+	#copy for concatentated run
+	S="$StudyFolder/$SessionCross/MNINonLinear/Results/$ConcatName"
+	T="$StudyFolder/$SessionLong/MNINonLinear/Results/$ConcatName"
+	local file files_to_copy="Movement_Regressors_demean.txt ReclassifyAsNoise.txt ReclassifyAsSignal.txt"
+	mkdir -p "$T"
+	for file in $files_to_copy; do
+		cp "$S"/"$file" "$T"/
+	done
+	
+	local ICADir="$S/${ConcatName}_hp$HighPass.ica"
+	local ICADirLong="$T/${ConcatName}_hp$HighPass.ica"
+	mkdir -p "$ICADirLong"				
+	if [ -f "$ICADir/HandNoise.txt" ]; then 
+		cp $ICADir/HandNoise.txt $ICADirLong/
```

Comment:
It would be nice if you quoted the rest of these variables in the command lines.
```suggestion
		cp "$ICADir"/HandNoise.txt "$ICADirLong"/
```

Reply:
Ok